geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStoreFactory.scala

```diff
@@ -216,4 +213,34 @@ object AccumuloDataStoreFactory extends GeoMesaDataStoreInfo {
 
     AuthorizationsProvider.apply(params, java.util.Arrays.asList(auths: _*))
   }
+
+  /**
+   * Configuration options for AccumuloDataStore
+   *
+   * @param catalog table in Accumulo used to store feature type metadata
+   * @param defaultVisibilities default visibilities applied to any data written
+   * @param generateStats write stats on data during ingest
+   * @param authProvider provides the authorizations used to access data
+   * @param audit optional implementations to audit queries
+   * @param queries query config
+   * @param writeThreads number of threads used for writing
+   */
+  case class AccumuloDataStoreConfig(
+      catalog: String,
+      defaultVisibilities: String,
+      generateStats: Boolean,
+      authProvider: AuthorizationsProvider,
+      audit: Option[(AuditWriter with AuditReader, AuditProvider, String)],
+      queries: AccumuloQueryConfig,
+      writeThreads: Int,
+      namespace: Option[String]
+    ) extends GeoMesaDataStoreConfig
+
+  case class AccumuloQueryConfig(
+      threads: Int,
+      recordThreads: Int,
```

Comment:
What's the difference between 'threads' and 'recordThreads'?

Reply:
recordThreads is for queries against the records/`id` table - we've always had that as a separate config in accumulo although nowhere else...